sdk/tables/Azure.Data.Tables/src/Extensions/TableEntityExtensions.cs

```diff
@@ -12,7 +12,7 @@ internal static class TableEntityExtensions
         /// The default case is intentionally unhandled as this means that no type annotation for the specified type is required.
         /// This is because the type is naturally serialized in a way that the table service can interpret without hints.
         /// </summary>
-        internal static IDictionary<string, object> ToOdataAnnotatedDictionary<T>(this T entity) where T : class, ITableEntity
+        internal static IDictionary<string, object> ToOdataAnnotatedDictionary<T>(this T entity)
```

Comment:
why remove `ITableEntity` here?

Reply:
It's not used. I can bring it back if you'd like.

Reply:
I don't feel strongly.